src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -212,6 +212,42 @@ bool ngraph::snippets::pass::PropagatePrecision::validate_and_infer_types_and_re
 
         if (output.get_element_type() != op_output_types[i]) {
             was_updated = true;
+
+            // option #1: if convertion after the operation exists then check if possible to remove it instead add new
+            const auto& target_inputs = output.get_target_inputs();
+            // limit by 1 child, while the issue was not resolved
+            // issue #107966
```

Comment:
Honestly I don't understand what is a problem with `output.get_target_inputs().size()`.
Let's to discuss offline

Reply:
extended

Reply:
I see that now the optimization works only these cases when all target inputs are supported `Convert` nodes. But I don't see the problems to support cases still when just some of target inputs (not all) are `Convert` nodes.

Just imagine.
We have `Node_0` with `0`-th output where 3 Nodes are connected: `target_inputs = {Node_1, Convert_0, Node_2}`

We iterate by `target_inputs` and verify each `target_input`: if we can remove Convert, we do it.
`Node_1` - isn't `Convert`
`Convert_0` - is `Convert` and call `can_be_removed` returns `true`. We remove this Convert from `target_inputs`:
```
// output.get_target_inputs().size() is increasing in line below
// as result we need to remove existing child_input manually
 // issue #107966
existing_convert->output(0).replace(output);
output.remove_target_input(target_input);
```
It's safe because we iterate by local set `target_inputs` (it's not changed), not by `node->get_output_target_inputs(0)` which can be changed during the optimization.

As for `Result input tensor renaming`, you can do it before `Convert` removing:
```
existing_convert_target_inputs = existing_convert->get_output_target_inputs(0);
for (const auto& input : existing_convert_target_inputs) {
      if (ngraph::is_type<ngraph::op::Result>(input.get_node())) {
            // Result input tensor name was changed, the name has to be restored
            // issue #107826
            input.get_tensor_ptr()->add_names(output.get_tensor_ptr()->get_names());
            op->set_friendly_name(existing_convert->get_friendly_name());
       }
}
```

And after renaming, you can call `Convert` removing (first code part).

What do you think?  Maybe I missed something.

Reply:
not actual more: reworked to support more cases and `Result` validation, thaks for comment